src/compositor/lipstickcompositor.cpp

```diff
@@ -700,9 +702,12 @@ void LipstickCompositor::setUpdatesEnabled(bool enabled)
                 m_onUpdatesDisabledUnfocusedWindowId = topmostWindow->windowId();
                 clearKeyboardFocus();
             }
-            hide();
-            if (QWindow::handle()) {
-                QGuiApplication::platformNativeInterface()->nativeResourceForIntegration("DisplayOff");
+            if (calledFromDBus()) {
+                setDelayedReply(true);
+                m_displayOffReply = message().createReply();
+                startTimer(1000);
```

Comment:
Both lipstick and the client can be delayed by stuff and not have a chance to respond there, so setting a low value runs the risk of the bug not being fixed. We're already emitting the displayAboutToBeOff, so the fade-to-black should have happened so the user doesn't notice the time too much. I think 1000ms is ok.